lib/pages/login_and_registration/basic_onboarding_page.dart

```diff
@@ -0,0 +1,209 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:runningapp/database/repository.dart';
+import 'package:runningapp/pages/logged_in/home_page/home_page.dart';
+
+final onboardingProvider =
+    StateNotifierProvider<OnboardingNotifier, OnboardingState>(
+  (ref) => OnboardingNotifier(),
+);
+
+class OnboardingState {
+  final String name;
+  final String age;
+  final String username;
+
+  OnboardingState({
+    this.name = '',
+    this.age = '',
+    this.username = '',
+  });
+}
+
+class OnboardingNotifier extends StateNotifier<OnboardingState> {
+  OnboardingNotifier() : super(OnboardingState());
+
+  void updateName(String name) {
+    state =
+        OnboardingState(name: name, age: state.age, username: state.username);
+  }
+
+  void updateAge(String age) {
+    state =
+        OnboardingState(name: state.name, age: age, username: state.username);
+  }
+
+  void updateUsername(String username) {
+    state =
+        OnboardingState(name: state.name, age: state.age, username: username);
+  }
+
+  Future<void> saveToFirestore() async {
+    final user = FirebaseAuth.instance.currentUser;
+    await FirebaseFirestore.instance.collection('users').doc(user!.uid).set({
+      'name': state.name,
+      'age': state.age,
+      'username': state.username,
+    }, SetOptions(merge: true));
+  }
+}
+
+class OnboardingPage extends ConsumerStatefulWidget {
+  const OnboardingPage({super.key});
+
+  @override
+  OnboardingPageState createState() => OnboardingPageState();
+}
+
+class OnboardingPageState extends ConsumerState<OnboardingPage> {
+  final PageController _pageController = PageController();
+
+  bool _isNameValid = false;
+  bool _isAgeValid = false;
+
+  void _validateName(String value) {
+    setState(() {
+      _isNameValid = value.trim().isNotEmpty;
+    });
+  }
+
+  void _validateAge(String value) {
+    setState(() {
+      _isAgeValid = value.isNotEmpty && int.tryParse(value) != null;
```

Comment:
Did we check if we disallow a negative age ? Not logic breaking but nice to have if we don't